modules/external_petsc_solver/src/base/ExternalPetscSolverApp.C

```diff
@@ -27,6 +30,13 @@ ExternalPetscSolverApp::ExternalPetscSolverApp(InputParameters parameters) : Moo
 
 #if LIBMESH_HAVE_PETSC
   PETScExternalSolverCreate(_comm->get(), &_ts);
+  DM da;
+  TSGetDM(_ts, &da);
+
+  DMCreateGlobalVector(da, &_petsc_sol);
+
+  // This requires because libMesh incorrectly treats the PETSc parallel vector as a ghost vector
```

Comment:
"This requires" -> "This is required"

Have you talked with @roystgnr about this incorrect behavior?

Reply:
I know what was the problem.  But I did not see an easy to fix it.

Reply:
And I do not want these change depend on another libmesh update.

Reply:
It's fair to not want to wait. But eventually if you think something is wrong at a lower level, then we should get it right at the lower level.

Reply:
I agreed. I added a PR here https://github.com/libMesh/libmesh/pull/2461. 

We do not need this line of code once libmesh is updated in MOOSE. 

But let us keep this line of code as is right now so that we can merge this change without waiting for another libmesh update.